HorizonEngine/src/HorizonEngine/Scripting/ScriptEngine.cpp

```diff
@@ -14,51 +17,25 @@ namespace Hzn
 
 		MonoAssembly* coreAssembly = nullptr;
 		MonoImage* coreAssemblyImage = nullptr;
+		std::filesystem::path coreAssemblyPath;
 
 		MonoAssembly* appAssembly = nullptr;
 		MonoImage* appAssemblyImage = nullptr;
+		std::filesystem::path appAssemblyPath;
+		bool reloadPending = false;
 	};
 
 	ScriptData* ScriptEngine::s_Data = nullptr;
 
 	namespace Utils
 	{
-		static char* ReadBytes(const std::string& filepath, uint32_t* outSize)
-		{
-			std::ifstream stream(filepath, std::ios::binary | std::ios::ate);
-
-			if (!stream)
-			{
-				// Failed to open the file
-				return nullptr;
-			}
-
-			std::streampos end = stream.tellg();
-			stream.seekg(0, std::ios::beg);
-			uint32_t size = end - stream.tellg();
-
-			if (size == 0)
-			{
-				// File is empty
-				return nullptr;
-			}
-
-			char* buffer = new char[size];
-			stream.read((char*)buffer, size);
-			stream.close();
-
-			*outSize = size;
-			return buffer;
-		}
-
 		static MonoAssembly* LoadCSharpAssembly(const std::string& assemblyPath)
 		{
 			uint32_t fileSize = 0;
-			char* fileData = Utils::ReadBytes(assemblyPath, &fileSize);
 
 			// NOTE: We can't use this image for anything other than loading the assembly because this image doesn't have a reference to the assembly
```

Comment:
Why is an image required?

Reply:
Image's are what hold the intermediate code, in the right formats which is then used to access information in the DLL, or the assembly.